pandas/core/base.py

```diff
@@ -1,10 +1,13 @@
 """
 Base and utility classes for pandas objects.
 """
+from typing import Any, Callable, Optional, Tuple, Union
+
 from pandas import compat
 from pandas.compat import builtins
 import numpy as np
 
+from pandas.types.hinting import ArrayLike, PythonScalar, Scalar, Buffer
```

Comment:
I am actually ok in doing

``from pandas.types.hinting import *  # noqa`` (you need the noqa because of the * import.

easier to type I think as well.